### DISCUSSION TITLE
Portable atomics?

### DISCUSSION THREAD

**Discussion by cirospaciari** [libuv/libuv#3391] [Ideas]
hello everyone, i was implementing refcounting and i'm using libuv in my project but i believe mutex is overkill for that, searching about atomic ops inside libuv i found this old Issue in the old repository:
https://github.com/joyent/libuv/issues/726
And this  proof-of-concept:
https://gist.github.com/bnoordhuis/3fafd09f65a789dd98bc

Any chance of atomics coming to libuv?

**Comment by bnoordhuis** [comment]
Probably not.

I wrote that when compiler support for C11 was minimal to non-existent but nowadays you just `#include <stdatomic.h>` and it'll work nearly everywhere.

**Comment by vtjnash** [comment]
Answered in https://github.com/libuv/libuv/issues/663
